app/src/main/java/server/RouteObject.java

```diff
@@ -0,0 +1,24 @@
+package server;
+
+import java.util.*;
+
+interface RouteInterface {
+    String getResponse(String HTTPVersion, String headers, String body);
+}
+
+public class RouteObject {
+    String route;
+    ArrayList<String> methods;
+    RouteInterface objectResponse;
+
+    public RouteObject(String objectRoute, ArrayList<String> objectHeaders, RouteInterface response){
+        this.route = objectRoute;
+        this.methods = objectHeaders;
+        this.objectResponse = response;
+    }
+
+    public String getObjectResponse(String HTTPVersion, String headers, String body){
+        return this.objectResponse.getResponse(HTTPVersion, headers, body);
```

Comment:
The naming here confuses me a little, specifically RouteInterface vs objectResponse. In one of your tests you call the lambda a "behavior". What about renaming `RouteInterface` to `RouteBehavior`?
Also changing `RouteObject` to just `Route` since we already know it's an object.
So then you have `new Route(name, methods, behavior)` in the RouteList, and then `this.behavior.getResponse(...` in this file.

Reply:
Makes sense to me. A lot of these variables were kind of done on the fly. This makes more sense.